package/MDAnalysis/core/universe.py

```diff
@@ -1640,23 +1649,32 @@ def guess_TopologyAttrs(
                     fg =  attr in force_guess
                     try:
                         values = guesser.guess_attr(attr, fg)
-                    except ValueError as e:
+                    except NoDataError as e:
                         if error_if_missing or fg:
                             raise e
                         else:
                             warnings.warn(str(e))
                             continue
 
-                    if values is not None:
-                        if attr in objects:
-                            self._add_topology_objects(
-                                attr, values, guessed=True)
-                        else:
-                            guessed_attr = _TOPOLOGY_ATTRS[attr](values, True)
-                            self.add_TopologyAttr(guessed_attr)
-                        logger.info(
-                            f'attribute {attr} has been guessed'
-                            ' successfully.')
+                    # None indicates no additional guessing was done
+                    if values is None:
+                        continue
+                    if attr in objects:
+                        # delete existing connections if they exist
+                        if fg and hasattr(self.atoms, attr):
+                            group = getattr(self.atoms, attr)
+                            self._delete_topology_objects(attr, group)
+                        # this method appends to existing bonds
```

Comment:
Something like that - i.e. I know what `_add_bonds` does, but reading it for the first time I'd be a bit lost as to what is meant here.

```suggestion
                        # this method appends any new bonds in values to existing bonds
```